### DISCUSSION TITLE
Why do internal packages with "main" and "types" pointing to "src/index.ts" also have TypeScript dependency and a tsconfig.json file?

### DISCUSSION THREAD

**Discussion by BenJackGill** [vercel/turborepo#6230] [Help] [answered]
I like the idea of pointing package.json `main ` and `types` to `src/index.ts` as championed by Turbopreo here: https://turbo.build/repo/docs/handbook/sharing-code/internal-packages

Mostly I like this because I can skip the build step for internal packages and changes also instantly appear throughout the monorepo.

But why does the guide (linked above) also include a `typescript` dependency and `tsconfig.json` file for the internal package?

If nothing is actually built in the internal package then why is that included?

The consuming app has it's own `typescript` dependency and `tsconfig.json` file to which the imported package would conform to when building / transpiling to JavaScript. So the inclusion of `typescript` dependency and `tsconfig.json` in the internal package file seems useless.

**Comment by anthonyshew** [comment, marked as answer]
The package itself needs TypeScript for your editor to be able to understand it. You can see what ends up happening by deleting the `tsconfig` and the dependency.

**Reply by BenJackGill** [reply]
I removed the `tsconfig` file and the TypeScript dependency from my internal packages a long time ago and it was fine. No IDE errors (Visual Studio Code). The editor understand everything correctly. Been coding without it for a while now. Which is why I brought up the confusing docs because it seemed like an oversight to keep it there.
